src/Response.php

```diff
@@ -89,12 +101,41 @@ private function getUpdateVersion() : array {
 			$this->updateSegment = random_int(0, 99);
 		}
 
+		$useNewerVersion = true;
+		if ($this->platform === 'win32') {
+			if ($this->osVersion === '10') {
+				switch ($this->kernelVersion) {
```

Comment:
@mgallien This looks complicated. Shouldn't we just split `10.0.xxxxx` into 3 components, and then, just check if the last part `xxxxx` is greater or equal to 1709?